components/ipfs/ipfs_json_parser.h

```diff
@@ -47,13 +48,15 @@ class IPFSJSONParser {
   static std::string RemovePeerFromConfigJSON(const std::string& json,
                                               const std::string& peer_id,
                                               const std::string& address);
+#if BUILDFLAG(ENABLE_IPFS_LOCAL_NODE)
   // Local pins
   static absl::optional<ipfs::AddPinResult> GetAddPinsResultFromJSON(
```

Comment:
We should probably guard header includes for these *PinResult structs too?

Reply:
We probably want to take it even further then, right? There are instances of including `ipfs_local_pin_service` for instance in the `brave_wallet` component

Reply:
Yes, for any headers that's only built when ENABLE_IPFS_LOCAL_NODE is true should be guarded with it. cc @cypt4

Reply:
Added guards to headers and moved the pin service in wallet to be behind buildflag